Backend/lib/sessionizer-metrics/src/Lib/SessionizerMetrics/EventStream.hs

```diff
@@ -0,0 +1,60 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+
+module Lib.SessionizerMetrics.EventStream where
+
+import Kernel.Prelude hiding (at, traceId)
+import Kernel.Utils.Common
+import Lib.SessionizerMetrics.Kafka.Internal
+import Lib.SessionizerMetrics.Types.Event
+
+triggerEvent ::
+  ( EventStreamFlow m r,
+    ToJSON p
+  ) =>
+  Event p ->
+  m ()
+triggerEvent event = do
+  allEventStream <- asks (.eventStreamMap)
+  let streamNames = filter (elem (eventType event) . eventTypes) allEventStream
+  logDebug $ "my filtered stream map" <> show streamNames
+  forM_ streamNames $ \stream -> do
+    case streamName stream of
+      KAFKA_STREAM -> do
+        let KafkaStream matchedConfig = stream.streamConfig
+        fork "updating in kafka" $ streamUpdates event matchedConfig
+      _ -> logDebug "Default stream"
+
+createEvent :: (MonadReader r1 m, MonadGuid m, MonadTime m, HasField "getDeploymentVersion" r2 Text, HasField "version" r1 r2) => Maybe Text -> Text -> EventType -> Service -> EventTriggeredBy -> Maybe p -> m (Event p)
+createEvent personId merchantId eventType service triggredBy payload = do
+  version <- asks (.version)
+  uid <- generateGUID
+  now <- getCurrentTime
+  let ev =
+        Event
+          { id = uid,
+            traceId = Nothing,
```

Comment:
why `traceId` and `sessionId` is nothing?
Similarly, what about primaryId and subType?